### ISSUE TITLE
Cart Termination and other skills that ignores caster's % damage cards not working properly on Pre-renewal

### ISSUE THREAD

**Issue by darhylism** [rathena/rathena#2577]
* **rAthena Hash**: c4a364b

* **Client Date**: 2015-11-04aRagexe

* **Server Mode**: Pre-Renewal

* **Description of Issue**: 
nk (skill damage properties) type 0x08 not working properly on pre-renewal

  * Result: 

06 nk (skill damage properties):
//    0x01 - No damage skill
//    0x02 - Has splash area
//    0x04 - Damage should be split among targets
//    0x08 - Skill ignores caster's % damage cards (misc type always ignores)
//    0x10 - Skill ignores elemental adjustments
//    0x20 - Skill ignores target's defense (misc type always ignores)
//    0x40 - Skill ignores target's flee (magic type always ignores)
//    0x80 - Skill ignores target's def cards

Skills:

Soul Destroyer
Cart Termination
Bulls Eye
Throw Venom Knife
Wall of Thorns
 
The listed can be affected by turtle general card or 'bonus2 bAddClass,Class_All' even if its set to nk = 0x08 on pre-renewal

  * Expected Result: 
Turtle general card, and other same effect cards should not affect those skills.

  * How to Reproduce: 
  * Official Information:

Skill Cart Termination official description:
ONLY cards that add ATK increase High Speed Cart Ram's damage. Race/Size (ignoring the ATK gained from Size cards) cards and Turtle General cards do not add any damage.

  

* **Modifications that may affect results**:

**Comment by mazvi** [comment]
Is there normal before you update to the last rathena version?

**Comment by darhylism** [comment]
it is functioning properly before the initial work of map server to C++

**Comment by mazvi** [comment]
can you provide information version of your GCC? and what is your operating system, bit, and version?

**Comment by darhylism** [comment]
Tested on windows 7 64 bit,

 and on my vps OS:
Linux 2.6.32-696.13.2.el6.x86_64 #1 SMP Thu Oct 5 21:22:16 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux


 GCC: (GCC) 5.3.1 20160406 (Red Hat 5.3.1-6)

**Comment by Jeybla** [comment]
Could be related to #2582

**Comment by darhylism** [comment]
tried the fix from [#2582](https://github.com/rathena/rathena/pull/2582) still not working properly

**Comment by aleos89** [comment]
Could you try finding `battle_skill_get_damage_properties` in `battle.cpp` and replace it with this:
```cpp
static int battle_skill_get_damage_properties(uint16 skill_id, int is_splash)
{
	int nk = skill_get_nk(skill_id);
	if( !skill_id && is_splash ) //If flag, this is splash damage from Baphomet Card and it always hits.
		nk |= NK_NO_CARDFIX_ATK|NK_IGNORE_FLEE;
	return nk;
}
```
